client/scss/components/_header.scss

```diff
@@ -108,44 +103,159 @@
 .header__nav {
   display: none;
   margin-top: 0.5rem;
+  position: absolute;
+  top: 100%;
+  left: map-get($container-padding, 'small');
+  right: map-get($container-padding, 'small');
+
+  @include respond-to('medium') {
+    left: map-get($container-padding, 'medium');
+    right: map-get($container-padding, 'medium');
+  }
 
   .no-js &,
   .header--is-burger-open & {
     display: block;
   }
 
-  @include respond-to('medium') {
+  @include respond-to('header-medium') {
+    position: static;
     display: block;
     margin-top: 0;
   }
 }
 
+.header__search {
+  flex: 1;
+  display: flex;
+  justify-content: flex-end;
+
+}
+
+.header__form {
+  position: relative;
+}
+
+.header__input-wrap {
+  position: absolute;
+  right: 100%;
+  margin-right: -1px;
+  top: 0;
+  bottom: 0;
+  width: 10rem;
+  display: none;
+
+  .no-js &,
+  .header__form--is-active & {
```

Comment:
I'm still more of a fan of setting the modifiers on the elements themselves... personal preference though.